gateway/gateway.go

```diff
@@ -59,9 +60,9 @@ func (g gateway) registerEndpoints() (*http.ServeMux, error) {
 			}
 		}
 		// strip prefix from testRequest URI, but leave the trailing "/"
-		mux.Handle(prefix, http.StripPrefix(prefix[:len(prefix)-1], gwmux))
+		g.mux.Handle(prefix, http.StripPrefix(prefix[:len(prefix)-1], gwmux))
```

Comment:
I am not sure I really understand why we need to leave trailing "/" here and how is it validated that it will be exactly "/"?
What if an user pass `/v1` as prefix? Will the `gmux` receive `/v`?
Could you please add comment that explain such need in order to help someone who probably will work on this code in some day?

Reply:
this is a good callout, and should probably be changed. There also weren't any tests associated with this `gateway.go` file, so more work to do there